### PR TITLE
Revamp the server info queries and add options

### PR DESCRIPTION
**Pull request by blogh**
- Divide the server info query in several sub queries:
  + server info (activities, guc and instance stats)
  + temporary file stats
  + wal sender stats
  + wal receiver stats
  + replication slots
  This change is made in order to reduce duplication of queries and
  has other benfits (see below).

- temporary file stats require special privileges and can take a
  very long time to fetch:
  + add a timeout (400ms) to the query
  + add a --no-tempfile to skip tempfile stats
  + --rds now implies --no-tempfile
  + add a mecanism to skip tempfile checks if we fail once because of
  a lack of privileges or timeout to avoid log spam.

- wal receiver stats cannot be fetched on aurora because the function
  is not implemented.
  + add a --no-walreceiver to skip walreceiver stats
  + add a mecanism to skip tempfile checks if we fail once because of
  a lack of privileges to avoid log spam.

- Update the man page
  + new options
  + add a "MISSING DATA IN THE UI ?" section

### CODE REVIEW

File: docs/man/pg_activity.pod [hunk 1]
```diff
@@ -445,6 +461,47 @@ See: https://www.postgresql.org/docs/current/libpq-envars.html
 
 =back
 
+=head1 MISSING DATA IN THE UI ?
```
Comment (dlax) [inline docs/man/pg_activity.pod:464]: No space before punctuation in English:
```suggestion
=head1 MISSING DATA IN THE UI?
```

File: docs/man/pg_activity.pod [hunk 2]
```diff
@@ -445,6 +461,47 @@ See: https://www.postgresql.org/docs/current/libpq-envars.html
...
+
+On some OS like MacOS, psutils.io_counters() is not implemented. The effect are
+the same as thoses described when data_directory is not readable.
+
+pg_activity needs to access the pgsql_tmp directory stored in all tablespaces
+in order to compute the number and size of the temporary files. This requires
+the usage of the pg_ls_tmpdir() function (or pg_ls_dir() and pg_stats_file()
+for versions older than PostgreSQL 12). The user needs to own the SUPERUSER
+privilege, be a member of pg_read_server_files or have EXECUTE rights on the
+function to fetch the information.  When the number of tempfile grows a lot,
```
Comment (dlax) [inline docs/man/pg_activity.pod:492]:
```suggestion
function to fetch the information.  When the number of tempfiles grows a lot,
```

File: docs/man/pg_activity.pod [hunk 3]
```diff
@@ -445,6 +461,47 @@ See: https://www.postgresql.org/docs/current/libpq-envars.html
...
+the query might also timeout. The first failure to access this data will Temp
+file stats can be completely skipped with B<--no-tempfile>.
```
Comment (dlax) [inline docs/man/pg_activity.pod:494]: Last sentence seems to be missing words or so.

File: docs/man/pg_activity.pod [hunk 4]
```diff
@@ -445,6 +461,47 @@ See: https://www.postgresql.org/docs/current/libpq-envars.html
...
+
+Aurora doesn't provide the pg_stat_get_wal_receiver() function.  Therefore
+there is no wal receiver data in the B<process & information HEADER> group. The
+first failure to access this data might be logged by the PostgreSQL, the
+following checks will be skipped.  Wal receiver checks can be completely
+disabled with B<--no-walreceiver>.
+
+Finally, some information are not available in older version of PostgreSQL,
```
Comment (dlax) [inline docs/man/pg_activity.pod:502]:
```suggestion
Finally, some information is not available in older version of PostgreSQL,
```

File: pgactivity/data.py [hunk 5]
```diff
@@ -146,6 +148,9 @@ def pg_is_local_access(self) -> bool:
         try:
             query = queries.get("get_pid_file")
             with self.pg_conn.cursor() as cur:
+                # This query doesn't crash when the user doesn't have the
+                # requiered privilege to acces the data_directory guc
```
Comment (dlax) [inline pgactivity/data.py:152]:
```suggestion
                # required privilege to access the data_directory guc
```

File: pgactivity/data.py [hunk 6]
```diff
@@ -186,13 +191,105 @@ def pg_terminate_backend(self, pid: int) -> bool:
             ret: Dict[str, bool] = cur.fetchone()
         return ret["is_stopped"]
 
-    DbInfoDict = Dict[str, Union[str, int, float]]
+    def pg_get_temporary_file(self) -> Optional[TempFileInfo]:
+        """
+        Count the number of temporary files and gets their total size
```
Comment (dlax) [inline pgactivity/data.py:196]:
```suggestion
        Count the number of temporary files and get their total size.
```

File: pgactivity/data.py [hunk 7]
```diff
@@ -186,13 +191,105 @@ def pg_terminate_backend(self, pid: int) -> bool:
...
+        else:
+            query = queries.get("get_temporary_files_oldest")
+
+        with self.pg_conn.cursor() as cur:
+            try:
+                cur.execute(queries.get("set_statement_timeout"), {"timeout": "400ms"})
+                cur.execute(query)
+                ret = cur.fetchone()
+            except InsufficientPrivilege:
+                # superuser or pg_read_server_files are requiered (Issue #278)
```
Comment (dlax) [inline pgactivity/data.py:216]:
```suggestion
                # superuser or pg_read_server_files are required (Issue #278)
```

File: docs/man/pg_activity.pod [hunk 8]
```diff
@@ -240,7 +240,7 @@ required by another session. It shows following information:
 
 =item B<--rds>
 
-	Enable support for AWS RDS.
+	Enable support for AWS RDS (implies --no-tempfiles).
```
[the remark this answers is not in the document]
Comment (blogh) [reply docs/man/pg_activity.pod:243]: I went with --no-tempfiles

File: pgactivity/cli.py [hunk 9]
```diff
@@ -43,7 +43,7 @@ def get_parser() -> ArgumentParser:
         "--rds",
         dest="rds",
         action="store_true",
-        help="Enable support for AWS RDS.",
+        help="Enable support for AWS RDS (implies --no-tempfiles).",
```
[the remark this answers is not in the document]
Comment (blogh) [reply pgactivity/cli.py:46]: I went with --no-tempfiles

File: pgactivity/cli.py [hunk 10]
```diff
@@ -62,6 +62,22 @@ def get_parser() -> ArgumentParser:
         help="Skip total size of DB.",
         default=False,
     )
+    # --no-tempfile
+    group.add_argument(
+        "--no-tempfile",
```
[the remark this answers is not in the document]
Comment (blogh) [reply pgactivity/cli.py:67]: agreed,  I went with `--no-tempfiles`  in the options and `notempfiles` in the code.

File: docs/man/pg_activity.pod (hunk 2 above)
Comment (blogh) [reply docs/man/pg_activity.pod:492]: Fixed

File: docs/man/pg_activity.pod (hunk 3 above)
Comment (blogh) [reply docs/man/pg_activity.pod:494]: Replaced with :

```
When the number of tempfiles grows a lot,
the query might also timeout. The first failure to access this data will 
disable tempfile statistics. The feature can be disabled with
B<--no-tempfiles>.
```

File: docs/man/pg_activity.pod (hunk 4 above)
Comment (blogh) [reply docs/man/pg_activity.pod:502]: fixed

File: pgactivity/data.py (hunk 5 above)
Comment (blogh) [reply pgactivity/data.py:152]: Fixed

File: pgactivity/data.py (hunk 6 above)
Comment (blogh) [reply pgactivity/data.py:196]: Fixed

File: pgactivity/data.py (hunk 7 above)
Comment (blogh) [reply pgactivity/data.py:216]: Fixed

File: docs/man/pg_activity.pod (hunk 1 above)
Comment (blogh) [reply docs/man/pg_activity.pod:464]: Fixed.

File: docs/man/pg_activity.pod [hunk 11]
```diff
@@ -445,6 +461,47 @@ See: https://www.postgresql.org/docs/current/libpq-envars.html
...
+restricted access.
+
+Some settings are visible only to superusers or members of pg_read_all_settings
+such as the data_directory guc. If the user cannot read this parameter or
+access the pid file in the PGDATA directory, the B<system information HEADER>
+group will not be displayed. The B<%CPU>, B<%MEM>, B<Read/s> and B<Write/s>
+columns will also be missing from the B<ACTIVITY PANEL>.
+
+On some OS like MacOS, psutils.io_counters() is not implemented. The effect are
+the same as thoses described when data_directory is not readable.
```
[the remark this answers is not in the document]
Comment (blogh) [reply docs/man/pg_activity.pod:485]: Fixed

**Pull request merged**